packages/fxa-auth-server/test/local/routes/subscriptions.js

```diff
@@ -2095,6 +2019,113 @@ describe('DirectStripeRoutes', () => {
+      const actual = await directStripeRoutesInstance.reactivateSubscription(
+        reactivateRequest
+      );
+
+      assert.isEmpty(actual);
+    });
+  });
+
+  describe('updateSubscription', () => {
+    describe('when the plan is a valid upgrade', () => {
+      it('returns the subscription id', async () => {
+        const subscriptionId = 'sub_123';
+        const expected = { subscriptionId: subscriptionId };
+
+        directStripeRoutesInstance.stripeHelper.subscriptionForCustomer.resolves(
+          subscription2
+        );
+        directStripeRoutesInstance.stripeHelper.verifyPlanUpgradeForSubscription.resolves();
+        directStripeRoutesInstance.stripeHelper.changeSubscriptionPlan.resolves();
+
+        sinon.stub(directStripeRoutesInstance, 'customerChanged').resolves();
+
+        VALID_REQUEST.params = { subscriptionId: subscriptionId };
+        VALID_REQUEST.payload = { planId: 'plan_123' };
+
+        const actual = await directStripeRoutesInstance.updateSubscription(
+          VALID_REQUEST
+        );
+
+        assert.deepEqual(actual, expected);
+      });
+    });
+
+    describe('when the orginal subscription is not found', () => {
+      it('throws an exception', async () => {
+        directStripeRoutesInstance.stripeHelper.subscriptionForCustomer.resolves();
+        VALID_REQUEST.params = { subscriptionId: 'sub_123' };
+        VALID_REQUEST.payload = { planId: 'plan_123' };
+
+        return directStripeRoutesInstance
+          .updateSubscription(VALID_REQUEST)
+          .then(
+            () => Promise.reject(new Error('Method expected to reject')),
+            err => {
+              assert.instanceOf(err, WError);
+              assert.equal(err.errno, error.ERRNO.UNKNOWN_SUBSCRIPTION);
+              assert.equal(err.message, 'Unknown subscription');
+            }
+          );
+      });
+    });
+  });
+
+  describe('listPlans', () => {
+    it('returns the available plans when auth headers are valid', async () => {
+      const expected = PLANS;
+
+      directStripeRoutesInstance.stripeHelper.allPlans.returns(PLANS);
+      const actual = await directStripeRoutesInstance.listPlans(VALID_REQUEST);
+
+      assert.deepEqual(actual, expected);
+    });
+
+    it('results in an error when auth headers are invalid', async () => {
+      const invalid_request = {
+        auth: {
+          credentials: {
+            scope: ['profile'],
+            user: `${UID}`,
+            email: `${TEST_EMAIL}`,
+          },
+        },
+      };
+
+      return directStripeRoutesInstance.listPlans(invalid_request).then(
+        () => Promise.reject(new Error('Method expected to reject')),
+        err => {
+          assert.instanceOf(err, WError);
+          assert.equal(err.message, 'Requested scopes are not allowed');
+        }
+      );
+    });
+  });
+
+  describe('listActive', () => {});
+
+  describe('getCustomer', () => {});
+
+  describe('sendSubscriptionStatusToSqs', () => {});
```

Comment:
These are coming later?

Reply:
I added placeholders for upcoming tests to reduce likelihood of merge conflicts